polaris-core/src/main/java/org/apache/polaris/core/persistence/transactional/TransactionalMetaStoreManagerImpl.java

```diff
@@ -195,6 +195,24 @@ private void dropEntity(
       ms.writeEntityInCurrentTxn(callCtx, entityGrantChanged, false, originalEntity);
     }
 
+    // delete all policy mapping records
+    try {
+      final List<PolarisPolicyMappingRecord> mappingOnPolicy =
+          (entity.getType() == PolarisEntityType.POLICY)
+              ? ms.loadAllTargetsOnPolicyInCurrentTxn(
+                  callCtx, entity.getCatalogId(), entity.getId())
+              : List.of();
+      final List<PolarisPolicyMappingRecord> mappingOnTarget =
+          (entity.getType() == PolarisEntityType.POLICY)
```

Comment:
Should we narrow down the entity type? I think entity types like principal, task, principal role may never be a policy target type? Then this query will be a net waste. How about we introduce a static method like `attachableEntityTypes()`, I guess it'd be useful somewhere else.